packages/performance/src/services/remote_config_service.ts

```diff
@@ -219,3 +267,30 @@ function configValid(expiry: string): boolean {
 function shouldLogAfterSampling(samplingRate: number): boolean {
   return Math.random() <= samplingRate;
 }
+
+/**
+ * True if event should be sent to Fl transport endpoint rather than log endpoint.
+ * rolloutPercent is in range [0.0, 100.0].
+ * @param {string} iid Installation ID which identifies a web app installed on client.
+ * @param {number} rolloutPercent the possibility of this app sending events to Fl endpoint.
+ * @return {boolean} true if this installation should send events to Fl endpoint.
+ */
+export function isDestFl(iid: string, rolloutPercent: number): boolean {
+  if (iid.length === 0) {
+    return false;
+  }
+  return getHashPercent(iid) < rolloutPercent;
+}
+/**
+ * Generate integer value range in [0, 99].
+ * @param {string} seed Same seed will generate consistent hash value using this algorithm.
+ * @return {number} Hash value in range [0, 99], generated from seed and hash algorithm.
+ */
+function getHashPercent(seed: string): number {
+  let hash = 0;
+  for (let i = 0; i < seed.length; i++) {
+    hash = (hash << 3) + hash - seed.charCodeAt(i);
```

Comment:
Added comment to call out the source.

Reply:
Am I missing something? I still don't see the reference link to the algorithm.

Reply:
This is at the function comment: `Implementation from String.hashCode() in Java.`. This will keep the comment concise and we can find reference online.